client/CServerHandler.cpp

```diff
@@ -489,7 +489,7 @@ void CServerHandler::sendMessage(const std::string & txt) const
 		std::string connectedId, playerColorId;
 		readed >> connectedId;
 		readed >> playerColorId;
-		if(connectedId.length(), playerColorId.length()) // BUG https://bugs.vcmi.eu/view.php?id=3144
+		if(connectedId.length() < playerColorId.length())
```

Comment:
reading the following code, I don't find the fix correct. It looks like the intention was to check that strings are not empty.

Reply:
Exactly, this command assigns specified player color to connectionId. It doesn't make any sense to compare string length. Passing connectionId="5" and playerColorId="1" would be absolutely valid but fails with proposed change

Reply:
Still the syntax is weird. Cpp ref does not say it is legal to put , inside if and what does it mean

Reply:
Looks like it is comma operator and according to it the first statement result will be just discurded

Reply:
> Looks like it is comma operator and according to it the first statement result will be just discurded

yes, you're absolutely correct

Reply:
@kambala-decapitator OK, then I will remove from this PR changes affected `client/CServerHandler.cpp`